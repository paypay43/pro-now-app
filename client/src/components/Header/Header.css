.sticky-top {
  position: sticky;
  top: 0;
  z-index: 1020;
  background-color: #4a5e83;
  border-bottom: 1px solid #222b3c;
}

.site-header {
}

.navbar-flex {
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-right: 1.5rem;
  margin-left: 1.5rem;
  font-size: 1.5rem;
  height: 50px;
}

.navbar-flex h3 {
  margin-right: auto;
  text-decoration: none;
  color: #ebebeb;
}

.navbar-flex a {
  text-decoration: none;
  color: white;
  margin-left: 0.5rem;
  font-size: 1.5rem;
  font-weight: bold;
}
.navbar-flex ul {
  list-style-type: none;
}
.navbar-flex ul li {
  margin: 0 1rem;
  display: inline;
}

.navbar-flex ul li a:hover {
  color: #c1c1c1;
}
