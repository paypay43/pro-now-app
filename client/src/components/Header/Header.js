import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import TokenService from '../../services/token-service';
import UserContext from '../../contexts/UserContext';
import './Header.css';

export default function Header(props) {
  const userContext = React.useContext(UserContext);
  //const [loggedIn, setLogged] = useState(userContext.loggedIn);

  const renderLogoutLink = () => {
    return (
      <ul className="logged-out">
        <li className="logged-in">
          <Link to="/login" onClick={handleLogout}>
            Logout
          </Link>
        </li>
      </ul>
    );
  };

  const handleLogout = () => {
    TokenService.clearAuthToken();
    userContext.setLogged(false);
  };

  const renderLoginLink = () => {
    return (
      <ul className="logged-out">
        <li>
          <Link to="/register">Register</Link>
        </li>
        <li>
          <Link to="/login">Log in</Link>
        </li>
      </ul>
    );
  };

  return (
    <UserContext.Consumer>
      {value => {
        return (
          <nav className="sticky-top">
            <div class="navbar-flex">
              <h3>
                <FontAwesomeIcon className="black" icon="fist-raised" />
                <Link to="/">ProtestNow</Link>
              </h3>
              {value.loggedIn ? renderLogoutLink() : renderLoginLink()}
            </div>
          </nav>
        );
      }}
    </UserContext.Consumer>
  );
}
