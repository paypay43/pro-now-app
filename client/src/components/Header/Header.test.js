import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

describe(`Header component`, () => {
  const props = {};

  it('renders header', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});
