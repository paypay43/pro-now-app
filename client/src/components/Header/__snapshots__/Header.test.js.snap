// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders header 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
